the functionality of the widget should be finished

gusfel opened the pull request:
- added some tests
- you can sort by star rating by clicking the star rating in the ratings window
- the "relevant" sort now comes from the server
- finished adding small functionality like marking review asa helpful

laenalea reviewed (approved):
Great job, can't wait to see the CSS come together!

Three little queries:
1) Will the modal exit option be added later on? If I open the modal, there's no way for me to click out of it unless I submit a completed form.

2) The relevance algorithm you came up with doesn't output the expected behavior when the reviews have the same helpfulness. I looked down to your reviews that have 0 helpfulness, sorted by relevance. The relevance of all equal helpfulness score reviews should show the most recent at the top, but it show the older reviews first.

3) This is more of a conceptual question than a suggestion. but should the reviews collapse back to show only two at a time once you switch to sort by a different category? (e.g. expanding all reviews, then switching to a diff category sort doesn't collapse the reviews)

momo1113 commented:
So much going on the page!

Not sure about the if the submit button next to the sorting works properly. When I sorted by newest, then click submit button. The reviews' section will disappear, is that suppose to be?

gusfel commented:
> Great job, can't wait to see the CSS come together!
> 
> Three little queries:
> 
> 1. Will the modal exit option be added later on? If I open the modal, there's no way for me to click out of it unless I submit a completed form.
> 2. The relevance algorithm you came up with doesn't output the expected behavior when the reviews have the same helpfulness. I looked down to your reviews that have 0 helpfulness, sorted by relevance. The relevance of all equal helpfulness score reviews should show the most recent at the top, but it show the older reviews first.
> 3. This is more of a conceptual question than a suggestion. but should the reviews collapse back to show only two at a time once you switch to sort by a different category? (e.g. expanding all reviews, then switching to a diff category sort doesn't collapse the reviews)

thanks for the feedack.  
1. i added a cancel button to the new review modal in my current version
2. based off of what rob said when he talked to us i've updated the relevant sort to just use the relevant sort from the api
3. yeah that would be a good idea to work on after getting everything to look right
